src/app.ts

```diff
@@ -0,0 +1,90 @@
+import { Probot, Context } from 'probot';
+
+interface PullRequestInfo {
+  owner: string
+  repo: string
+  pull_number: number
+}
+
+const getPRInfo = async (context: Context, {
+  owner, repo, pull_number
+}: PullRequestInfo) => {
+  const {
+      data: {
+        commits,
+        head
+      }
+    } = await context.octokit.pulls.get({
+      owner, repo, pull_number
+    });
+  
+  return { prNumCommits: commits, prHeadSha: head.sha, prBranchName: head.ref }
+}
+
+const createEmptyCommit = async (
+  context: Context,
+  {
+    owner,
+    repo,
+    prHeadSha
+  }: PullRequestInfo & { prHeadSha: string }
+) => {
+  const { data: { sha } } = await context.octokit.git.createCommit({
+        owner,
+        repo,
+        message: 'empty commit',
+        tree: '', //TODO To check how to get the SHA of the tree object this commit points to
+        parent: prHeadSha
+  })
+  
+  return sha
+}
+
+const pushCommitToBranch = (
+  context: Context,
+  {
+    owner,
+    repo,
+    branchName,
+    commitSha
+  }: PullRequestInfo & { branchName: string, commitSha: string},
+) => {
+  return context.octokit.git.createRef({
+        owner,
+        repo,
+        ref: branchName,
+        sha: commitSha
+      })
+}
+
+module.exports = (app: Probot): void => {
+  app.log("Yay! The app was loaded!");
+
+  //TODO pull_request.opened should be an event too?
```

Comment:
Would it be possible a user opens a PR, gets reviewed (or no reviewer is required) and it tries to merge the PR. In this case, `pull_request.edited` event would not trigger right?

Reply:
right

Reply:
I just assume the app needs to run on:
- `pull_request.opened`: First time a PR is opened
- `pull_request.synchronize`: Everytime the head changes.
---
The """problem""" is that it will run twice everytime an empty commit is done.

Reply:
you mean it will get triggered by itself? That's okay, we can check the author of the commit is the app itself, and ignore it in that case